### PR TITLE
[@types/saturn-web] Add type for Saturn SD (https://saturnhq.io)

### PR DESCRIPTION
**Pull request by wrongsahil**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/saturn-web/index.d.ts
1. types/saturn-web/saturn-web-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 20eac1eca35dc9bb7fe6daad16982332dcebff10
</p>

**Review by rohanmathur91 — APPROVED** [review]

**Comment by jakebailey** [comment]
Do you have any links to docs that would show that this PR is correct?

File: types/saturn-web/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        integrationId: string;
+        isOpen: boolean;
+        isLoaded: boolean;
+        isUserLoaded: boolean;
+        setUser: (
+            identifier: string,
+            user: {
+                email: string;
+                name?: string;
+                identifierHash?: string;
```
Comment (jakebailey) [inline types/saturn-web/index.d.ts:20]:
```suggestion
                name?: string | undefined;
                identifierHash?: string | undefined;
```

For `exactOptionalPropertyTypes`.

**Review by jakebailey — COMMENTED** [review]
It appears as though this is not public, so nobody can really verify the accuracy of these types...

**Comment by peterblazejewicz** [comment]
I'd say, there is no real gain for OSS/TS users atm IMO, this is behind-waiting-list only service at the moment

**Comment by wrongsahil** [comment]
Hey @jakebailey. Sorry for the delay. You can find the docs for the SDK [here](https://developers.saturnhq.io/web-sdk/installation/js-sdk).
Incase it helps, i can share a non-minified version of the SDK so that you can verify it. 

@peterblazejewicz Yes the codebase is closed source. But lots of people are using Saturn for their projects and struggling with the type definition errors. We've temporarily release the types [here] for now (https://www.npmjs.com/package/@saturnhq/types) but this requires extra configurations which is a pain.

**Comment by weswigham** [comment]
AFAIK, we've decided to merge this, but @jakebailey has an unresolved comment up above.

**Review by sheetalkamat — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Comment by wrongsahil** [comment]
Ready to merge

**Comment by wrongsahil** [comment]
I got a little bit busy with other things. So wasn't able to look into this. 

Thanks everyone for the reviews. 🙌

**Pull request merged**